controllers/src/main/java/com/redhat/ipaas/controllers/WorkflowHandler.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.controllers;
+
+import com.redhat.ipaas.model.integration.Integration;
+
+import java.util.Set;
+
+/**
+ *
+ */
+public interface WorkflowHandler {
+
+    Set<Integration.Status> getTriggerStatuses();
+
+    Integration.Status execute(Integration model) throws Exception;
```

Comment:
I would ditch the `throws Exception` as we wrap all exception under IPaaSException.